File: javascript/Array/arrayMesh2/arrayMesh2.js
// Given ONE array of strings, return a new array that contains every combination of each string with every other string in the array.

// Input: ["a", "b", "c", "d"]
// Output: ["ab", "ac", "ad", "ba", "bc", "bd", "ca", "cb", "cd", "da", "db", "dc"]

const arrayMeshTwo = function (array) {
  let result = []

  for (let i = 0; i < array.length; i += 1) {
    for (let j = 0; j < array.length; j += 1) {
      if (array[i] != array[j]) {
        result.push(array[i] + array[j])
      }
    }
  }

  return result
}

console.log(arrayMeshTwo(["a", "b", "c", "d"]))

File: ruby/algo_ladder/array_mesh_2/array_mesh_2.rb
# Given ONE array of strings, return a new array that contains every combination of each string with every other string in the array.

# Input: ["a", "b", "c", "d"]
# Output: ["ab", "ac", "ad", "ba", "bc", "bd", "ca", "cb", "cd", "da", "db", "dc"]

def array_mesh_2(input_array)
  result = []
  index_one = 0
  while index_one < input_array.length
    index_two = 0
    while index_two < input_array.length
      if index_one != index_two
        result << input_array[index_one] + input_array[index_two]
      end
      index_two += 1
    end
    index_one += 1
  end
  return result
end

p array_mesh_2(["a", "b", "c", "d"])